File: code/example_03/python/testme.py
#!/usr/bin/python
import holder

myobj = holder.Object();

m = "rusty is a cowboy!";

#myobj.setText(None);
myobj.setText(m);

r = myobj.getText();

if m != r: raise AsertionError

print "m and r should be the same:"
print "     m=", m
print "     r=", r

myholder = holder.ObjectHolder();

try:
  #myholder.setObject("hiya");
  myholder.setObject(myobj);
except TypeError as te:
  print("bad type! %s" % te)
except Exception as e:
  print("unknown reason fail! %s" % e)

other = myholder.getObject();
if other.getText() != r: raise AsertionError

File: code/example_03/perl/testme.pl
#!/usr/bin/perl
use holder;
use strict;
use warnings;

#################### test Object #########################

my $myobj = holder::Object->new();

my $m = "rusty is a cowboy";
#my $m = <>; chomp $m;

#$myobj->setText(0);
$myobj->setText($m);

my $r = $myobj->getText();

die unless "$m" eq "$r";

print("\$m and \$r should be the same:\n\t\$m = $m\n\t\$r = $r\n");

#################### test ObjectHolder #########################

my $myholder = holder::ObjectHolder->new();
die if "$myholder" eq "";

#$myholder->setObject("hiya");
$myholder->setObject($myobj);

my $other = $myholder->getObject();
die unless $other->getText() eq $r;

